app/experimenter/experiments/tests/test_tasks.py

```diff
@@ -373,3 +374,104 @@ def test_internal_only_does_not_add_comment(self):
+        ExperimentFactory.create_with_status(
+            target_status=Experiment.STATUS_ACCEPTED, normandy_id=1234
+        )
+
+        self.mock_normandy_requests_get.return_value = (
+            self.buildMockSuccessDisabledResponse()
+        )
+        tasks.update_experiment_status()
+        updated_experiment = Experiment.objects.get(normandy_id=1234)
+        self.assertEqual(updated_experiment.status, Experiment.STATUS_ACCEPTED)
+        self.assertFalse(
+            updated_experiment.changes.filter(
+                old_status=Experiment.STATUS_ACCEPTED,
+                new_status=Experiment.STATUS_LIVE,
+            ).exists()
+        )
+
+    def test_live_experiment_stays_live_if_normandy_enabled(self):
+        ExperimentFactory.create_with_status(
+            target_status=Experiment.STATUS_LIVE, normandy_id=1234
+        )
+        tasks.update_experiment_status()
+        updated_experiment = Experiment.objects.get(normandy_id=1234)
+        self.assertEqual(updated_experiment.status, Experiment.STATUS_LIVE)
+        self.assertFalse(
+            updated_experiment.changes.filter(
+                old_status=Experiment.STATUS_LIVE,
+                new_status=Experiment.STATUS_COMPLETE,
+            ).exists()
+        )
+
+    def test_live_experiment_becomes_complete_if_normandy_disabled(self):
+        ExperimentFactory.create_with_status(
+            target_status=Experiment.STATUS_LIVE, normandy_id=1234
+        )
+        self.mock_normandy_requests_get.return_value = (
+            self.buildMockSuccessDisabledResponse()
+        )
+
+        tasks.update_experiment_status()
+        updated_experiment = Experiment.objects.get(normandy_id=1234)
+        self.assertTrue(
+            updated_experiment.changes.filter(
+                old_status=Experiment.STATUS_LIVE,
+                new_status=Experiment.STATUS_COMPLETE,
+            ).exists()
+        )
+
+    def test_one_failure_does_not_affect_other_experiment_status_updates(self):
+        self.setUpMockNormandyFailWithSpecifiedID("1234")
+        ExperimentFactory.create_with_status(
+            target_status=Experiment.STATUS_ACCEPTED, normandy_id=1234
+        )
+
+        ExperimentFactory.create_with_status(
+            target_status=Experiment.STATUS_ACCEPTED, normandy_id=1235
+        )
+
+        tasks.update_experiment_status()
+        updated_experiment = Experiment.objects.get(normandy_id=1234)
+        updated_experiment2 = Experiment.objects.get(normandy_id=1235)
+        self.assertFalse(
+            updated_experiment.changes.filter(
+                old_status=Experiment.STATUS_ACCEPTED,
+                new_status=Experiment.STATUS_LIVE,
+            ).exists()
+        )
+        self.assertTrue(
+            updated_experiment2.changes.filter(
+                old_status=Experiment.STATUS_ACCEPTED,
+                new_status=Experiment.STATUS_LIVE,
+            ).exists()
+        )
```

Comment:
Also it occurs to me we should add a test case where there's an unrecognized email in the normandy data and a user is created with that email